starknet-ff/src/lib.rs

```diff
@@ -1,56 +1,166 @@
+
+        let mut res = U256::ZERO;
+        for b in value.bytes().map(|b| b.wrapping_sub(b'0')) {
+            if b > 9 {
+                return Err(FromDecStrError::InvalidCharacter);
+            }
+            let r = {
+                let product = res.checked_mul(&U256::from_u8(10));
+                if product.is_some().into() {
+                    product.unwrap()
+                } else {
+                    return Err(FromDecStrError::OutOfRange);
+                }
+            };
+            let r = {
+                let sum = r.checked_add(&U256::from_u8(b));
+                if sum.is_some().into() {
+                    sum.unwrap()
+                } else {
+                    return Err(FromDecStrError::OutOfRange);
+                }
+            };
+            res = r;
+        }
+
+        Fp256::<FrParameters>::from_repr(BigInteger256::new(res.to_uint_array()))
+            .map(|inner| Self { inner })
+            .ok_or(FromDecStrError::OutOfRange)
+    }
+
+    pub fn from_hex_be(value: &str) -> Result<Self, FromHexError> {
+        let value = value.trim_start_matches("0x");
+
+        let hex_chars_len = value.len();
+        let expected_hex_length = U256_BYTE_COUNT * 2;
+
+        let parsed_bytes: [u8; U256_BYTE_COUNT] = if hex_chars_len == expected_hex_length {
+            let mut buffer = [0u8; U256_BYTE_COUNT];
+            hex::decode_to_slice(value, &mut buffer).map_err(|_| FromHexError::InvalidCharacter)?;
+            buffer
+        } else if hex_chars_len < expected_hex_length {
+            let mut padded_hex = str::repeat("0", expected_hex_length - hex_chars_len);
+            padded_hex.push_str(value);
+
+            let mut buffer = [0u8; U256_BYTE_COUNT];
+            hex::decode_to_slice(&padded_hex, &mut buffer)
+                .map_err(|_| FromHexError::InvalidCharacter)?;
+            buffer
+        } else {
+            return Err(FromHexError::InvalidLength);
+        };
+
+        match Self::from_bytes_be(&parsed_bytes) {
+            Ok(value) => Ok(value),
+            Err(_) => Err(FromHexError::OutOfRange),
+        }
     }
 
     /// Attempts to convert a big-endian byte representation of a field element into an element of
-    /// this prime field. Returns None if the input is not canonical (is not smaller than the
+    /// this prime field. Returns error if the input is not canonical (is not smaller than the
     /// field's modulus).
     ///
     /// ### Arguments
     ///
-    /// * `bytes`: The byte array in **big endian** format
-    pub fn from_bytes_be(bytes: &[u8; 32]) -> Option<Self> {
-        let mut bits = [false; 32 * 8];
-        for (ind_byte, byte) in bytes.iter().enumerate() {
-            for ind_bit in 0..8 {
-                bits[ind_byte * 8 + ind_bit] = (byte >> (7 - ind_bit)) & 1 == 1;
-            }
+    /// * `bytes`: The byte slice in **big endian** format
+    pub fn from_slice_be(bytes: &[u8]) -> Result<Self, FromByteSliceError> {
```

Comment:
Do you think it's necessary to have this function? In the lib itself it's used only once. I can't judge how useful it is for the lib uses, but maybe just having `from_bytes_be` is ok?

Reply:
I think we should keep the public API surface as small as possible as removing functions is a breaking change. Lemme remove this for now.

Before this I could have just made it `pub(crate)`, but now since `FieldElement` is a standalone crate, I have to start thinking carefully about the interface I guess.

Reply:
Done. Replaced the occurrence with an `unsafe` block instead.